pkg/sql/table.go

```diff
@@ -821,12 +821,15 @@ func (p *planner) findTableContainingIndex(
 	result = nil
 	for i := range tns {
 		tn := &tns[i]
-		tableDesc, err := MustGetTableDesc(
-			ctx, p.txn, p.getVirtualTabler(), tn, true, /*allowAdding*/
-		)
+		tableDesc, err := p.getTableDesc(ctx, tn)
```

Comment:
getTableDesc doesn't assert that it's a table (as opposed to a sequence or view); `MustGetTableDesc` does. That assertion is where the error was coming from.